node/nodeRunner.go

```diff
@@ -157,230 +158,258 @@ func printEnableEpochs(configs *config.Configs) {
 }
 
 func (nr *nodeRunner) startShufflingProcessLoop(
-	chanStopNodeProcess1 chan endProcess.ArgEndProcess,
+	chanStopNodeProcess chan endProcess.ArgEndProcess,
 ) error {
-	configs := nr.configs
-	flagsConfig := configs.FlagsConfig
-	configurationPaths := configs.ConfigurationPathsHolder
 	for {
-		goRoutinesNumberStart := runtime.NumGoroutine()
-
 		log.Debug("\n\n====================Starting managedComponents creation================================")
 
-		log.Debug("creating core components")
-		managedCoreComponents, err := nr.CreateManagedCoreComponents(
-			chanStopNodeProcess1,
-		)
-		if err != nil {
+		shouldStop, err := nr.executeOneComponentCreationCycle(chanStopNodeProcess)
+		if shouldStop {
 			return err
 		}
 
-		log.Debug("creating crypto components")
-		managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-		if err != nil {
-			return err
-		}
+		log.Debug("node statistics before running GC", statistics.GetRuntimeStatistics()...)
```

Comment:
can you extract the debug data here into a separate function and make this optional

Reply:
done, made optional through config flags